packages/desktop-client/src/components/modals/CoverModal.tsx

```diff
@@ -0,0 +1,115 @@
+import React, { useCallback, useEffect, useState } from 'react';
+import { useDispatch } from 'react-redux';
+
+import { pushModal } from 'loot-core/client/actions';
+
+import { useCategories } from '../../hooks/useCategories';
+import { useInitialMount } from '../../hooks/useInitialMount';
+import { styles } from '../../style';
+import { addToBeBudgetedGroup } from '../budget/util';
+import { Button } from '../common/Button';
+import { Modal } from '../common/Modal';
+import { View } from '../common/View';
+import { FieldLabel, TapField } from '../mobile/MobileForms';
+import { type CommonModalProps } from '../Modals';
+
+type CoverModalProps = {
+  modalProps: CommonModalProps;
+  categoryId: string;
+  onSubmit: (categoryId: string) => void;
+};
+
+export function CoverModal({
+  modalProps,
+  categoryId,
+  onSubmit,
+}: CoverModalProps) {
+  const { grouped: originalCategoryGroups, list: categories } = useCategories();
+  const categoryGroups = addToBeBudgetedGroup(
+    originalCategoryGroups.filter(g => !g.is_income),
+  );
+
+  const [fromCategoryId, setFromCategoryId] = useState<string | null>(null);
+  const dispatch = useDispatch();
+
+  const onCategoryClick = useCallback(() => {
+    dispatch(
+      pushModal('category-autocomplete', {
+        categoryGroups,
+        onSelect: categoryId => {
+          setFromCategoryId(categoryId);
+        },
+      }),
+    );
+  }, [categoryGroups, dispatch]);
+
+  const _onSubmit = (categoryId: string | null) => {
+    if (categoryId) {
+      onSubmit?.(categoryId);
+    }
+
+    modalProps.onClose();
+  };
+
+  const initialMount = useInitialMount();
+
+  useEffect(() => {
+    if (initialMount) {
+      onCategoryClick();
+    }
+  }, [initialMount, onCategoryClick]);
+
+  const category = categories.find(c => c.id === categoryId);
+
+  if (category == null) {
```

Comment:
<!-- Nitpicks are small, trivial, but necessary changes. Distinguishing nitpick comments significantly helps direct the reader's attention to comments requiring more involvement.  -->
**🥜 nitpick:** 

```suggestion
  if (category === null) {
```